### PR TITLE
Simplifying SPLS implementation. Creating levelized SPLS.

### PR DESCRIPTION
**Pull request by wdhawkins**
This PR simplifies the SPLS implementation and creates a levelized version of SPLS. The levelized version is necessary to enable parallelism along the sweep front. Although not included in this PR, I have implemented and tested a levelized version of the AAH sweepchunk and all regression tests pass.

**Note that this PR changes our sweep ordering.**

### CODE REVIEW

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc [hunk 1]
```diff
@@ -60,12 +60,12 @@ AAH_FLUDSCommonData::InitializeAlphaElements(const SPDS& spds,
...
 
   // csoi = cell sweep order index
-  so_cell_inco_face_face_category_.reserve(spls.item_id.size());
-  so_cell_outb_face_slot_indices_.reserve(spls.item_id.size());
-  so_cell_outb_face_face_category_.reserve(spls.item_id.size());
-  for (int csoi = 0; csoi < spls.item_id.size(); ++csoi)
+  so_cell_inco_face_face_category_.reserve(spls.size());
+  so_cell_outb_face_slot_indices_.reserve(spls.size());
+  so_cell_outb_face_face_category_.reserve(spls.size());
+  for (int csoi = 0; csoi < spls.size(); ++csoi)
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc:66]: `size()` gives back `size_t` which is unsigned, but `int` is `signed`. So we should be changing that as we bump into these things.

The same problem is in the rest of this PR.

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc [hunk 2]
```diff
@@ -60,12 +60,12 @@ AAH_FLUDSCommonData::InitializeAlphaElements(const SPDS& spds,
...
   {
-    int cell_local_id = spls.item_id[csoi];
+    int cell_local_id = spls[csoi];
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc:68]: Please, change the type to `auto`. The for loops below can use that too. In general, I thinking, using `auto` is better...

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc (hunk 1 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc]: All of FluDS needs a rewrite, but I think I got most of them in this file.

**Review by andrsd — APPROVED** [review]

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/spds/aah.cc [hunk 3]
```diff
@@ -50,15 +50,38 @@ AAH_SPDS::AAH_SPDS(int id, const Vector3& omega, const MeshContinuum& grid, bool
...
+  if (spls_.empty())
   {
     throw std::logic_error("AAH_SPDS: Cyclic dependencies found in the local cell graph.\n"
                            "Cycles need to be allowed by the calling application.");
   }
 
+  // Generate levelized spls
+  int max_level = 0;
+  std::vector<int> levels(num_vertices(local_cell_graph), 0);
+  for (auto it = spls_.begin(); it != spls_.end(); ++it)
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/spds/aah.cc:65]: I noticed, that you could do: `for (auto & v: spls_) {...` now. And you don't need the `auto v = *it;` any more.

Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/spds/aah.cc]: Fixed.

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc [hunk 4]
```diff
@@ -152,10 +152,10 @@ AAH_FLUDSCommonData::SlotDynamics(const Cell& cell,
...
         for (auto cyclic_dependency : spds.LocalSweepFAS())
         {
-          int a = cyclic_dependency.first;
-          int b = cyclic_dependency.second;
-          int c = cell.local_id;
-          int d = face.GetNeighborLocalID(grid);
+          auto a = cyclic_dependency.first;
+          auto b = cyclic_dependency.second;
+          auto c = cell.local_id;
+          auto d = face.GetNeighborLocalID(grid);
```
Comment (ragusa) [inline modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc:158]: a,b,c,d?? the variable names do not seem helpful. was there a reason for this?

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc [hunk 5]
```diff
@@ -347,30 +347,30 @@ AAH_FLUDSCommonData::LocalIncidentMapping(const Cell& cell,
...
 
-    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Incident face
+    // Incident face
     if (orienation == FaceOrientation::INCOMING)
     {
       if (face.IsNeighborLocal(grid))
       {
         // Find associated face for dof mapping
-        int ass_face = cell_nodal_mapping[f].associated_face_;
+        auto ass_face = cell_nodal_mapping[f].associated_face_;
```
Comment (ragusa) [inline modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc:361]: here's another ass face.  `assoc_face`  might be preferred?

**Review by ragusa — APPROVED** [review]

**Review by zhardy-lanl — APPROVED** [review]

File: modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc (hunk 5 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/fluds/aah_fluds_common_data.cc:361]: Fixed.

**Pull request merged**